src/pybind/mgr/volumes/volume.py

```diff
@@ -0,0 +1,151 @@
+import json
+import errno
+import logging
+import orchestrator
+
+log = logging.getLogger(__name__)
+
+class VolumeClient(object):
+    def __init__(self, mgr):
+        """
+        """
+        self.mgr = mgr
+
+    def gen_pool_names(self, volname):
+        """
+        """
+        return "cephfs.{}.meta".format(volname), "cephfs.{}.data".format(volname)
+
+    def get_fs(self, fs_name):
+        fs_map = self.mgr.get('fs_map')
+        for fs in fs_map['filesystems']:
+            if fs['mdsmap']['fs_name'] == fs_name:
+                return fs
+        return None
+
+    def get_mds_names(self, fs_name):
+        fs = self.get_fs(fs_name)
+        if fs is None:
+            return []
+        return [mds['name'] for mds in fs['mdsmap']['info'].values()]
+
+    def volume_exists(self, volname):
+        return self.get_fs(volname) is not None
+
+    def create_pool(self, pool_name, pg_num, pg_num_min=None, pg_autoscale_factor=None):
+        # create the given pool
+        command = {'prefix': 'osd pool create', 'pool': pool_name, 'pg_num': pg_num}
+        if pg_num_min is not None:
+            command['pg_num_min'] = pg_num_min
+        r, outb, outs = self.mgr.mon_command(command)
+        if r != 0:
+            return r, outb, outs
+
+        # set pg autoscale if needed
+        if pg_autoscale_factor is not None:
```

Comment:
if not pg_autoscale_factor:

Reply:
Sorry, this should be if pg_autoscale_factor:

Reply:
already took care of that ;)